src/ShapeCrawler/Presentations/IFileProperties.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+
+#pragma warning disable IDE0130
+namespace ShapeCrawler;
+#pragma warning restore IDE0130
+
+/// <summary>
+///     Represents the core properties of the presentation file.
+/// </summary>
+/// <remarks>
+///     These properties are not related to the presentation itself, but rather the containing file.
+/// </remarks>
+public interface IFileProperties
+{
+    /// <summary>
+    ///     Gets or sets the category.
+    /// </summary>
+    string? Category { get; set; }
+
+    /// <summary>
+    ///     Gets or sets the type of content represented, generally defined by a specific use and intended audience. Example values include "Whitepaper", "Security Bulletin", and "Exam". (This property is distinct from MIME content types as defined in RFC 2045.)
+    /// </summary>
+    string? ContentType  { get; set; }
+
+    /// <summary>
+    ///     Gets or sets the status of the content. Example values include "Draft", "Reviewed", and "Final".
+    /// </summary>
+    string? ContentStatus { get; set; }
+
+    /// <summary>
+    ///     Gets or sets the creation date and time.
+    /// </summary>
+
+    DateTime? Created { get; set; }
```

Comment:
There is definitely a case for user being able to set the value of these somehow. If I create a new presentation in code, I do want to be able to declare who was responsible for creating it. In fact, this field is one of the first I'd actually use in production.

For `Modified` date there is a case for setting the date (of course), but there is also a case for NOT setting the date. One of the things I use ShapeCrawler for is tuning hand-created decks to increase their precision. In this case, the `Modified` date should be retained, as philosophically the modification I care about is the user's last modification. Likewise, I wouldn't really want to touch `Modified By` in this case either.

If you really want them moved to presentation saving, that does seem like a lot of extra data movement for not an obvious benefit, but will do it that way if needed.

For `Created` would you want the date/time automatically set when saved? If so, this will add in more complexity for testing, as we'll have to be able to control the date for tests. This a common pattern, so not really hard, but again adding complexity for not obvious benefit.

Reply:
Ok. I agree with these setters.

However, I still think that `Modified` should be updated also when user calls `IPresentationProperties.Save()`.

Reply:
2ebf84808d33c9d751f1fb44eab031b0093971d6 et al

Also set created/modified when creating a presentation.

Open to ideas on where the `ITimeProvider` should live.

Reply:
I would suggest creating a separate internal class `ShapeCrawler` and move this property there:

```C#
internal class ShapeCrawler
{
    internal static ITimeProvider TimeProvider { get; set; } = new SystemTimeProvider();
}
```

I previously wanted to use such a class for global settings when I wanted to collect some statistics with the user's permission `public bool CollectStatistics { get; set; } = false;` and send it to an external API. Perhaps we will also come back to this idea in the future.

Reply:
I would suggest creating a separate internal class `ShapeCrawler` and move this property there:

```C#
internal class ShapeCrawler
{
    internal static ITimeProvider TimeProvider { get; set; } = new SystemTimeProvider();
}
```

I previously wanted to use such a class for global settings when I wanted to collect some statistics with the user's permission `public bool CollectStatistics { get; set; } = false;` and send it to an external API. Perhaps we will come back to this idea in the future.

Reply:
Ok this is done in 901cbc3214e2e1c87ace88a658c056a001331679. Accidentally conflated that change and a style change in the same commit 😜

For the instrumentation feature, I suggest write up an issue with a basic design to get started.

Azure Log Analytics is $0.50/GB/mo for warm tier data. https://azure.microsoft.com/en-us/pricing/details/monitor/